specification/netapp/resource-manager/Microsoft.NetApp/stable/2021-06-01/netapp.json

```diff
@@ -3040,6 +3146,42 @@
         }
       }
     },
+    "SubscriptionQuotaItemList": {
+      "description": "List of Subscription Quota Items",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "A list of SubscriptionQuotaItems",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SubscriptionQuotaItem"
+          }
+        }
+      }
+    },
+    "SubscriptionQuotaItem": {
```

Comment:
>SubscriptionQuotaItem [](http://example.com/codeflow?start=5&length=21)

Why isn't this a standard ARM resource payload. You should have `id`, `name`, `type` properties and a `properties` property at a minimum.

Reply:
That is why I am wondering as this is not strictly a "resource" if a POST to list the quota values would be a better fit here?

`"/subscriptions/{subscriptionId}/providers/Microsoft.NetApp/locations/{location}/quotaLimits": {
    "post": {
      "tags": [
        "NetApp Resource"
      ],
      "summary": "Get quota limits",
      "description": "Get the default and current limits for quotas",
      "operationId": "NetAppResourceQuotaLimits_List",
      "parameters": [
        {
          "$ref": "#/parameters/SubscriptionId"
        },
        {
          "$ref": "#/parameters/location"
        },
        {
          "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
        }
      ],
      "responses": {
        "200": {
          "description": "OK",
          "schema": {
            "$ref": "#/definitions/SubscriptionQuotaItemList"
          }
        },
        "default": {
          "description": "Error response describing why the operation failed."
        }
      },
      "x-ms-examples": {
        "QuotaLimits": {
          "$ref": "examples/QuotaLimits_List.json"
        }
      }
    }
  },`

If that does not fit this scenario I will make those changes you suggest.

Reply:
Hi, @majastrz can you take another look?

Reply:
The problem with POST is that identities with Reader role aren't able to invoke them, which causes some frustration. It's also harder to discover POST actions like that. My suggestion would be to stick with GET and just adjust the payloads to match ARM resources. It should be a pretty minor change.

Reply:
@majastrz  yes I was thinking it used along with `authorizationActionMappings`.  
Note we have updated GET in the spec with regards to your comment.